src/v2/view-builder/utils/ChallengeViewUtil.js

```diff
@@ -98,17 +100,35 @@ export function doChallenge(view) {
     view.add(View.extend({
       className: 'app-link-content',
       template: hbs`
-            {{i18n code="appLink.content" bundle="login"}}
-          `
+        <div class="spinner"></div>
+        <div class="appLinkContent {{hideClass}}">{{i18n code="appLink.content" bundle="login"}}</div>
+      `,
+      getTemplateData() {
+        return { hideClass: HIDE_CLASS };
+      },
+      postRender() {
+        setTimeout(_.bind(()=> {
+          const data = { label: loc('goback', 'login') };
+          this.options.appState.trigger('updateFooterLink', data);
```

Comment:
Is there any way to figure this out based on IDX response ? instead of doing it in `postRender` ?

Reply:
Not really. The change is purely an UX change. The general flow still follows what's instructed by the IDX response. Only that we display a spinner first and delay displaying the corresponding UI immediately. This is done to buy time for the Okta Verify app to launch.
The footer link/action is still coming from the IDX response. In the spinner view, we use the cancel link which comes from `authenticatorChallenge.cancel` that does the same thing as when polling fails. In the later view, the link is from `cancel`. This applies to the assurance view as well.